Sources/AsyncQueue/ActorQueue.swift

```diff
@@ -80,65 +78,88 @@ public final class ActorQueue {
+        // Crashing here means that this queue is being sent tasks either before an execution context has been set, or
+        // after the execution context has deallocated. An ActorQueue's execution context should be set in the adopted
+        // actor's `init` method, and the ActorQueue should not exceed the lifecycle of the adopted actor.
+        let executionContext = self.executionContext! // Capture/retain the executionContext before suspending.
+        return await withUnsafeContinuation { continuation in
+            taskStreamContinuation.yield(ActorTask(executionContext: executionContext) { executionContext in
+                continuation.resume(returning: await task(executionContext))
+            })
         }
     }
 
     /// Schedules an asynchronous throwing task and returns after the task is complete.
     /// The scheduled task will not execute until all prior tasks have completed or suspended.
-    /// - Parameter task: The task to enqueue.
+    /// - Parameter task: The task to enqueue. The task's parameter is a reference to the actor whose execution context has been adopted.
     /// - Returns: The value returned from the enqueued task.
-    public func await<T>(_ task: @escaping @Sendable () async throws -> T) async throws -> T {
-        try await withUnsafeThrowingContinuation { continuation in
-            taskStreamContinuation.yield {
+    public func await<T>(_ task: @escaping @Sendable (isolated ActorType) async throws -> T) async throws -> T {
+        // Crashing here means that this queue is being sent tasks either before an execution context has been set, or
+        // after the execution context has deallocated. An ActorQueue's execution context should be set in the adopted
+        // actor's `init` method, and the ActorQueue should not exceed the lifecycle of the adopted actor.
+        let executionContext = self.executionContext! // Capture/retain the executionContext before suspending.
+        return try await withUnsafeThrowingContinuation { continuation in
+            taskStreamContinuation.yield(ActorTask(executionContext: executionContext) { executionContext in
                 do {
-                    continuation.resume(returning: try await task())
+                    continuation.resume(returning: try await task(executionContext))
                 } catch {
                     continuation.resume(throwing: error)
                 }
-            }
+            })
         }
     }
 
     // MARK: Private
 
-    private let taskStreamContinuation: AsyncStream<@Sendable () async -> Void>.Continuation
+    private let taskStreamContinuation: AsyncStream<ActorTask>.Continuation
 
-    // MARK: - ActorExecutor
+    /// The executionContext actor on whose isolated context our tasks run.
+    /// It is safe to use `unowned` here because it is API misuse to interact with an `ActorQueue` from an instance other than the `executionContext`.
+    private unowned var executionContext: ActorType?
```

Comment:
Hmm why not use `weak`? It's been awhile since I thought about `unowned` but I thought `unowned` was useful if you wanted a non-optional variable. Since this is already optional is there any downside to using `weak`?

Reply:
Entirely fair! 18a0183